src/game/hat.ts

```diff
@@ -169,13 +169,13 @@ registerEventHandler("hat-pickup", {
   },
 });
 
-registerEventHandler("hat-drop", {
+registerEventHandler<vec3>("hat-drop", {
```

Comment:
u shouldn't need this annotation as it should infer that from ur runEvent  or serialize/deserialze fns

Reply:
I actually get type errors without it--without this annotation Typescript doesn't seem to know which overload of registerEventHandler it's calling, so it needs type annotations on all of the arguments to the functions we're passing.

```
src/game/tool.ts:102:14 - error TS7006: Parameter 'em' implicitly has an 'any' type.

102   runEvent: (em, entities, location: vec3) => {
                 ~~

src/game/tool.ts:102:18 - error TS7006: Parameter 'entities' implicitly has an 'any' type.

102   runEvent: (em, entities, location: vec3) => {
                     ~~~~~~~~

src/game/tool.ts:114:20 - error TS7006: Parameter 'buf' implicitly has an 'any' type.

114   serializeExtra: (buf, location) => {
                       ~~~

src/game/tool.ts:114:25 - error TS7006: Parameter 'location' implicitly has an 'any' type.

114   serializeExtra: (buf, location) => {
                            ~~~~~~~~

src/game/tool.ts:117:22 - error TS7006: Parameter 'buf' implicitly has an 'any' type.

117   deserializeExtra: (buf) => {
                         ~~~
```